src/entry.rs

```diff
@@ -747,7 +747,27 @@ self.body_as_struct_sequence::<memory::platform_tuning::ElementRef<'_>>() {
                     let v = parameters.collect::<Vec<_>>();
                     state.serialize_field("parameters", &v)?;
                 } else {
-                    state.serialize_field("struct_body", &buf)?;
+                    match self.context.mem_dfe_search_version() {
```

Comment:
I suspect some of these cases can be simplified with judicious use of guards in the match arms.
https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html

Reply:
How? The idea is that the struct_* calls will return None if they can't tell how to handle it. Otherwise, it will do the cast and return an appropriate instance as Some(s).

I'd have to do something like bind something to parts of the match guard, no?

Reply:
I was thinking more along the lines of,

```
    match self.context.mem_dfe_search_version() {
        Some(MemDfeSearchVersoin::Genoa2) if let Some(s) = ... {
```

But yeah, as you noticed, `if let` guards are not yet stable.  Regardless, one could do something like:

```
    if let (Some(s), name) = match self.context.mem_dfe_search_version() {
        Some(MemDfeSearchVersion::Genoa2) => (self.body_as_struct_array::<memory::MemDfeSearchElement32>(), "MemDfeSearchElement32"),
        Some(MemDfeSearchVersion::Turin1) => (self.body_as_struct_array::<memory::MemDfeSearchElement36>(), "MemDfeSearchElement36"),
        _ => (None, ""),
    } {
        let v = s.iter().colect::<Vec<_>>();
        state.serialize_field(name, &v)?;
    } else {
        state.serialize_field("struct_body", &buf)?;
    }
```

Reply:
The types of the results of the respective body_as_struct_array calls are different.

Reply:
Ah, ok. Well, that's unfortunate.